### DISCUSSION TITLE
Getting milestone 2 completed (2h30m)

### DISCUSSION THREAD

**Discussion by g105b** [SuperHyperInstantFutureTime/TrackShift#35] [Work log]
I'm going to spend the next hour or so getting the last points to [Milestone 2](https://github.com/SuperHyperInstantFutureTime/Trackshift/milestone/2) accomplished.

**Comment by g105b** [comment]
Simple Behat tests to show working with a list of uploads:

https://github.com/SuperHyperInstantFutureTime/Trackshift/blob/9721d611233e8295c9672b21986f2a4f38e26462/test/behat/features/manage-uploads.feature

```behat
Feature: App should list out and remove uploaded files
	In order to retain control over my uploaded files
	As an uploader
	I should be able to manage the list of uploaded files on my account

	Scenario: I can see uploads list
		Given I am on the homepage
		Then I should see "Uploaded files (0)"

		When I attach the file "prs-simple-3-songs.csv" to "statement"
		And I press "Upload"
		Then I should see "Uploaded files (1)"
		And I should see "prs-simple-3-songs - 299 B - PRS Statement"

		When I attach the file "prs-simple-3-songs-another-statement.csv" to "statement"
		And I press "Upload"
		Then I should see "Uploaded files (2)"
		And I should see "prs-simple-3-songs-another-statement - 273 B - PRS Statement"
		# Existing file should still be present!
		And I should see "prs-simple-3-songs - 299 B - PRS Statement"

		When I attach the file "gubbins.txt" to "statement"
		And I press "Upload"
		Then I should see "Uploaded files (3)"
		And I should see "gubbins - 2.2 KB - Unknown"
		# Existing file should still be present!
		And I should see "prs-simple-3-songs-another-statement - 273 B - PRS Statement"
		And I should see "prs-simple-3-songs - 299 B - PRS Statement"

	Scenario: I can delete an individual upload
		Given I am on the homepage
		When I attach the file "prs-simple-3-songs.csv" to "statement"
		And I press "Upload"
		And I attach the file "prs-simple-3-songs-another-statement.csv" to "statement"
		And I press "Upload"
		Then I should see "Uploaded files (2)"
		When I press "Delete prs-simple-3-songs"
		Then I should see "Uploaded files (1)"
```

**Reply by g105b** [reply]
Now onto implementing these new tests.

**Comment by g105b** [comment]
![file-uploads](https://user-images.githubusercontent.com/358014/203777393-fb74152a-8e42-4350-89a5-b71ded0642c1.gif)

What's lovely about this is that it's just using the HTML file input. I added a few lines of JavaScript to display a message while dragging, but that's just superficial - the raw HTML is more than capable of processing multiple file uploads at once.

**Comment by g105b** [comment]
Working on automatic purges after 3 weeks. Just been experimenting with the best way of doing this - this is what it looks like so far:

![image](https://user-images.githubusercontent.com/358014/203784921-7bca6d4c-a733-4e09-ad52-b0050c98966d.png)

**Comment by richardbirkin** [comment]
@g105b this is great progress
